### ISSUE TITLE
Some tests for this package fail in sound null-safety mode

### ISSUE THREAD

**Issue by brianrobles204** [draw-dev/DRAW#204]
Hello,

I was using this package in my app just fine, but I noticed when I fully migrated my app to null safety and turned on `sound mode` that I was burned by this (now fixed) issue #203. I was curious why tests didn't catch this and it turns out that tests for this package are running in unsound mode because of the [reply dev dependency](https://pub.dev/packages/reply).

I tried creating a [fork of reply with null safety](https://github.com/brianrobles204/reply.dart), and after running the tests with full sound null-safety on, I notice there are still some lingering casting errors that cause tests to fail.

Tests in unsound mode:
```
DRAW % flutter test --no-sound-null-safety
Warning: You are using these overridden dependencies:                   
! reply 0.1.2-dev from git https://github.com/brianrobles204/reply.dart.git at faabf6
Running "flutter pub get" in DRAW...                                7.4s
00:04 +84: /Users/brianrobles204/Projects/Flutter/DRAW/test/redditor/redditor_test.dart: lib/redditor/gildings              
  Skip: Rewrite to handle new gilding types
00:05 +97 ~1: /Users/brianrobles204/Projects/Flutter/DRAW/test/user_content_moderation/user_content_moderation_test.dart: lib/user_content_moderation/suggested-sort
  Skip: Needs updating to support "CommentSortType.best"
00:06 +182 ~2: All tests passed!
```

Tests in sound mode:
```
DRAW % flutter test                       
00:04 +84: /Users/brianrobles204/Projects/Flutter/DRAW/test/redditor/redditor_test.dart: lib/redditor/gildings              
  Skip: Rewrite to handle new gilding types
00:05 +96 ~1: /Users/brianrobles204/Projects/Flutter/DRAW/test/user_content_moderation/user_content_moderation_test.dart: lib/user_content_moderation/suggested-sort
  Skip: Needs updating to support "CommentSortType.best"
00:05 +103 ~2 -1: /Users/brianrobles204/Projects/Flutter/DRAW/test/subreddit/subreddits_test.dart: lib/subreddits/stream [E]
  type 'Null' is not a subtype of type 'String'
  package:draw/src/util.dart 84:17  streamGenerator
  
00:05 +103 ~2 -1: loading /Users/brianrobles204/Projects/Flutter/DRAW/test/subreddit/subreddits_test.dart                   
Consider enabling the flag chain-stack-traces to receive more detailed exceptions.
For example, 'dart test --chain-stack-traces'.
00:05 +110 ~2 -2: /Users/brianrobles204/Projects/Flutter/DRAW/test/subreddit/subreddit_test.dart: lib/subreddit/comment_stream [E]
  type 'Null' is not a subtype of type 'String'
  package:draw/src/util.dart 84:17         streamGenerator
  ===== asynchronous gap ===========================
  test/subreddit/subreddit_test.dart 44:5  main.<fn>
  
00:05 +110 ~2 -2: loading /Users/brianrobles204/Projects/Flutter/DRAW/test/subreddit/subreddit_test.dart                    
Consider enabling the flag chain-stack-traces to receive more detailed exceptions.
For example, 'dart test --chain-stack-traces'.
00:05 +125 ~2 -3: /Users/brianrobles204/Projects/Flutter/DRAW/test/subreddit/subreddit_moderation_test.dart: lib/subreddit/subreddit_settings_local_modify [E]
  type 'Null' is not a subtype of type 'String'
  package:draw/src/models/subreddit_moderation.dart 376:34  SubredditSettings.description
  test/subreddit/subreddit_moderation_test.dart 279:21      main.<fn>
  
00:05 +125 ~2 -3: loading /Users/brianrobles204/Projects/Flutter/DRAW/test/subreddit/subreddit_moderation_test.dart         
Consider enabling the flag chain-stack-traces to receive more detailed exceptions.
For example, 'dart test --chain-stack-traces'.
00:05 +179 ~2 -3: Some tests failed.
```

The offending lines are easy enough to fix, but longer-term I think the reply dependency needs to be handled so that tests always run in sound null-safety mode.

**Comment by SupremeDeity** [comment]
Do you want to create a **PR** for this?  

> but longer-term I think the reply dependency needs to be handled so that tests always run in sound null-safety mode.
I think its fine as long the tests run on sound null safety by default. 

Before 2.x.x, this repo really needs a better documentation for some of its functions and complete sound null safety.

**Comment by brianrobles204** [comment]
Well the original `reply` dependency seems to be abandoned, the GitHub repo is archived. I could publish my null-safe fork and submit a PR for this package so it uses the forked dev dependency

The alternative is to rewrite the test code to completely drop the `reply` dependency, but that's more involved

Is the first way acceptable?

**Comment by SupremeDeity** [comment]
I unfortunately cannot give a exact answer to this. The actual author of this repo is busy with travel and as such gets little time to reply. We can only wait for now.

**Comment by bkonyi** [comment]
Thanks for your patience! It's been busy for me lately... :-(

Yes, `reply` is unfortunately abandoned by the original author (it was never even explicitly migrated to Dart 2.0) but moving off of the package would be quite painful. I think it's probably worth pulling in a fork to a new `third_party/reply` directory and migrating it to support null safety. I wouldn't expect that to be too difficult to do since it's quite a small package, and that would definitely be easier to do than migrating all the test over to mockito.

@brianrobles204 would you be willing to pull your fork into this repo under `third_party/reply` and update the pubspec to use a local dependency?

**Comment by brianrobles204** [comment]
Hello, I submitted a PR with the forked `reply` dev dependency in a `third_party` folder as suggested by @bkonyi :)

The tests should fail now since they should be running in sound mode. I can prepare a PR to fix the exceptions unless @SupremeDeity wants to

**Comment by SupremeDeity** [comment]
If you want to then go ahead, otherwise i will run the tests and fix the exceptions after i am done with #205

**Comment by SupremeDeity** [comment]
This can be closed.
